services/registration/src/routes/statistics.ts

```diff
@@ -2,98 +2,116 @@
 import { asyncHandler } from "@api/common";
 import express from "express";
 
-import { ApplicationModel } from "../models/application";
+import { ApplicationModel, StatusType } from "../models/application";
 import { BranchModel, BranchType } from "../models/branch";
 
 export const statisticsRouter = express.Router();
 
 statisticsRouter.route("/").get(
   asyncHandler(async (req, res) => {
-    let totalUsers = 0;
-    let confirmedUsers = 0;
-    const submittedUsers = await ApplicationModel.countDocuments({
-      applicationSubmitTime: { $exists: true, $ne: null },
-    });
-    let applications = {};
-    let confirmations = {};
+    const aggregatedUsers = await ApplicationModel.aggregate([
+      {
+        $group: {
+          _id: "$status",
+          count: { $sum: 1 },
+        },
+      },
+    ]);
 
-    const applicationBranchAggregator = [
+    const aggregatedApplicationData = await ApplicationModel.aggregate([
       {
         $group: {
           _id: "$applicationBranch",
+          branchName: { $first: "$applicationBranch.name" },
+          data: { $push: "$applicationData" },
+        },
+      },
+    ]);
+
+    const aggregatedBranches = await BranchModel.aggregate([
+      {
+        $group: {
+          _id: "$name",
           count: { $sum: 1 },
+          type: { $first: "$type" },
         },
       },
-    ];
-    const confirmationBranchAggregator = [
+    ]);
+
+    const aggregatedRejections = await BranchModel.aggregate([
+      {
+        $match: { status: StatusType.DENIED },
+      },
```

Comment:
Can you add a few comments throughout this handler that explains what information you're getting at each step and what it means? Sometimes, using aggregates can be a little opaque, so this would help other people when they're reading it later.